bench/macro/lsm-tree-bench-wp8.hs

```diff
@@ -392,20 +391,20 @@ doDryRun gopts opts = do
 doDryRun' :: GlobalOpts -> RunOpts -> IO ()
 doDryRun' gopts opts = do
     -- calculated some expected statistics for generated batches
-    id $ do
```

Comment:
the block makes new scope for `n`, `d` etc. See warnings below. It's  the same trick as 

```c
int main() {

   {
     // new scope in C
     
   }
```

though in C++ one can also use it for RAII.

---

Maybe the `id` is not needed, so you could write just `

```
doDryRun' gopts opts = do

   do
       -- inner scope
       
```

Reply:
I applied your suggestions. Thanks for elucidating "nested `do`-block" usage. Now that I'm better informed, I think you are right to leave the scoping limitation. I added a comment on the matter in case other eyes find the code confusing.